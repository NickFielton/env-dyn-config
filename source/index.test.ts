describe('index', () => {
    beforeEach(() => {
        jest.resetModules()
    })

    test.each(['development', 'test', 'production'])('nominal-%s', (place) => {
        // Prepare the environment
        process.env.NODE_ENV = place

        // Import the config
        const config = require('./__mocks__/configs/config')

        // Check config
        expect(config).toMatchSnapshot()
    })

    test('no-env given', () => {
        // Prepare the environment
        process.env.NODE_ENV = 'place'

        // Import the config
        const config = require('./__mocks__/configs/config')

        // Check config
        expect(config).toMatchSnapshot()
    })

    test('err no-env-names defined', () => {
        // Mock env-names.type
        jest.mock('./env-names.type', () => ({ default: {} }))

        // Import the config
        expect(() => require('./__mocks__/configs/config')).toThrow()
    })
})
