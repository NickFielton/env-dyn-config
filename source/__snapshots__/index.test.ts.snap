// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index no-env given 1`] = `
Object {
  "default": Object {
    "baseUrl": "https://dev.example.com",
    "endpoint": "/api/main",
    "general": "Only for devs",
  },
}
`;

exports[`index nominal-development 1`] = `
Object {
  "default": Object {
    "baseUrl": "https://dev.example.com",
    "endpoint": "/api/main",
    "general": "Only for devs",
  },
}
`;

exports[`index nominal-production 1`] = `
Object {
  "default": Object {
    "baseUrl": "https://example.com",
    "endpoint": "/api/main",
    "general": "Only for clients",
    "shared": "this property is shared with prod only",
  },
}
`;

exports[`index nominal-test 1`] = `
Object {
  "default": Object {
    "baseUrl": "https://dev.example.com",
    "endpoint": "/api/main",
    "general": "Only for tests",
  },
}
`;
